### PR TITLE
Properties with expression bodies don't add backing fields

### PR DESCRIPTION
**Pull request by jcouv**
### Customer scenario
Define a partial struct with a field and a property that isn't an auto-property. If the property uses expression-bodied accessors, a warning is reported as if it was an auto-property (ie. it has a backing field), when it actually isn't.

### Bugs this fixes
Fixes https://github.com/dotnet/roslyn/issues/23668

### Workarounds, if any
Suppress warning or use blocks in property bodies.

### Performance impact
None. Checking both for block and expression bodies where we used to only check for block bodies.

### Is this a regression from a previous update?
No.

### How was the bug found?
Reported by customer.

Note: this warning (CS0282) doesn't seem to have a VB counter-part. I'm 
```VB
' No warning
Partial Structure C
	Property x As Integer
End Structure
Partial Structure C
	Property y As Integer
End Structure
```

### CODE REVIEW

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for review. Tiny fix

File: src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowDiagnosticTests.cs [hunk 1]
```diff
@@ -1056,6 +1056,31 @@ partial struct S2
                 );
         }
 
+        [Fact]
+        [WorkItem(23668, "https://github.com/dotnet/roslyn/issues/23668")]
+        public void Bug()
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowDiagnosticTests.cs]: Consider providing a descriptive name. Perhaps `PartialWithPropertyButNoField`.

Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowDiagnosticTests.cs]: Slipped through :-S

File: src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowDiagnosticTests.cs [hunk 2]
```diff
@@ -1056,6 +1056,31 @@ partial struct S2
                 );
         }
 
+        [Fact]
+        [WorkItem(23668, "https://github.com/dotnet/roslyn/issues/23668")]
+        public void PartialWithPropertyButSingleField()
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowDiagnosticTests.cs]: >PartialWithPropertyButSingleField [](start = 20, length = 33)

It doesn't feel like this test belongs in this file. The CS0282 warning is not reported by flow analysis.

Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/FlowAnalysis/FlowDiagnosticTests.cs]: Moved it to SymbolErrorTests.cs where the other test for that warning is.

**Comment by jcouv** [comment]
test ubuntu_14_debug_prtest  please

**Comment by jcouv** [comment]
@MeiChin-Tsai for ask-mode approval for 15.6. Thanks

**Comment by MeiChin-Tsai** [comment]
approved. thx.

**Pull request merged**